### ISSUE TITLE
Edge of terrain casting weird shadow

### ISSUE THREAD

**Issue by GreaterThen0** [Facepunch/sbox-issues#7193]
### Describe the bug

Around the edge of my terrain, a weird shadow is being cast.

### To Reproduce

spawn a terrain chunk

feed it a height map

weird edge shadow

### Expected behavior

for the edge of the terrain to not cast this weird shadow

### Media/Files

![Image](https://github.com/user-attachments/assets/e39ecadc-b6c0-46d7-812f-2b63d88d2b1d)
![Image](https://github.com/user-attachments/assets/81e01b1d-fb24-4f21-a5a0-df2202693fc8)
![Image](https://github.com/user-attachments/assets/170edc7c-b012-4e57-8737-f6944f615023)

Video: https://www.youtube.com/watch?v=9m9MVXZe1tc

https://github.com/user-attachments/assets/91a895a9-de47-4ee4-9820-45cad74ac276

### Additional context

I believe the root cause is that the terrain doesn't have a right-angle edge and that the terrain is sampling outside the height map which ends up applying a bevelled edge as seen in one of the above pictures

**Comment by GreaterThen0** [comment]
With the help of DoctorGurke in the discord I found the source of the problem the function in the TerrainCommon.HLSL Terrain_Normal it uses this sampler

SamplerState g_sBilinearBorder < Filter( BILINEAR ); AddressU( BORDER ); AddressV( BORDER ); >;

by scraping the code out and changing to

SamplerState g_sSampler1 < Filter( ANISO ); AddressU( Wrap ); AddressV( Wrap ); >;

it appears to be fixed this should probably be updated


https://www.youtube.com/watch?v=wWdDWG-zTrU

**Comment by aylaylay** [comment]
Yeah we should look at this and clean up the edges, that geometry on the edges isn't desirable

**Comment by GreaterThen0** [comment]
> With the help of DoctorGurke in the discord I found the source of the problem the function in the TerrainCommon.HLSL Terrain_Normal it uses this sampler
> 
> SamplerState g_sBilinearBorder < Filter( BILINEAR ); AddressU( BORDER ); AddressV( BORDER ); >;
> 
> by scraping the code out and changing to
> 
> SamplerState g_sSampler1 < Filter( ANISO ); AddressU( Wrap ); AddressV( Wrap ); >;
> 
> it appears to be fixed the TerrainCommon.HLSL should probably be updated
> 
> https://www.youtube.com/watch?v=wWdDWG-zTrU

this appears to not have properly fixed the weird lighting problem

I still think the root cause is the bevelled edges of the terrain

**Comment by JustusHanschmann** [comment]
Hey @aylaylay are there any Updates on this, it still seems to behave the Same way (creating the weird bevels and shadows etc) using the bultin Noisefields and applying them to Terrain.Storage.HeightMap
